tests/runner.py

```
@@ -0,0 +1,308 @@
+                if c.exists():
+                    obfucc = str(c)
+                    break
+            # default fallback path
+            if not obfucc:
+                obfucc = str(Path('build')/OBFUCC_BIN)
+
+    # If obfucc not found on PATH or via --obfucc, require the build step to produce it.
+    # This keeps the harness consistent: it will prefer a built test binary under build/
+    # (e.g. build/Release/obfucc.exe on Windows) or the explicit --obfucc override.
+    if not Path(obfucc).exists() and shutil.which(obfucc) is None:
+        print(f"INFO: obfucc not found on PATH and no --obfucc provided; attempting to build the test binary into 'build/'.")
```

Ruff (0.13.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)